cartographer/mapping_3d/sparse_pose_graph.cc

```diff
@@ -423,6 +423,34 @@ void SparsePoseGraph::RunFinalOptimization() {
           .max_num_iterations());
 }
 
+void SparsePoseGraph::CalculateHistograms() {
+  translational_residual_ = common::Histogram();
```

Comment:
Any reason we keep these histograms around? Seems they have to be recomputed every time, so we could make this a local variable?